x/auth/keeper/deterministic_test.go

```diff
@@ -324,54 +248,37 @@ func (suite *DeterministicTestSuite) TestGRPCQueryBech32Prefix() {
 			types.NewModuleAddress("gov").String(),
 		)
 
-		suite.runBech32PrefixIterations(ak, prefix)
+		queryClient := suite.createAndReturnQueryClient(ak)
+		req := &types.Bech32PrefixRequest{}
+		testdata.DeterministicIterations(suite.ctx, suite.Require(), req, queryClient.Bech32Prefix, 0, true)
 	})
 
-	suite.runBech32PrefixIterations(suite.accountKeeper, "cosmos")
-}
-
-func (suite *DeterministicTestSuite) runAddressBytesToStringIterations(addressBytes []byte, prevRes string) {
-	for i := 0; i < iterCount; i++ {
-		res, err := suite.queryClient.AddressBytesToString(suite.ctx, &types.AddressBytesToStringRequest{
-			AddressBytes: addressBytes,
-		})
-
-		suite.Require().NoError(err)
-		suite.Require().NotNil(res)
-
-		suite.Require().Equal(res.AddressString, prevRes)
-	}
+	req := &types.Bech32PrefixRequest{}
+	testdata.DeterministicIterations(suite.ctx, suite.Require(), req, suite.queryClient.Bech32Prefix, 0, false)
 }
 
 func (suite *DeterministicTestSuite) TestGRPCQueryAddressBytesToString() {
 	rapid.Check(suite.T(), func(t *rapid.T) {
 		address := testdata.AddressGenerator(t).Draw(t, "address-bytes")
-		suite.runAddressBytesToStringIterations(address.Bytes(), address.String())
-	})
 
-	suite.runAddressBytesToStringIterations(addr.Bytes(), addr.String())
-}
-
-func (suite *DeterministicTestSuite) runStringToAddressBytesIterations(addressString string, prevRes []byte) {
-	for i := 0; i < iterCount; i++ {
-		res, err := suite.queryClient.AddressStringToBytes(suite.ctx, &types.AddressStringToBytesRequest{
-			AddressString: addressString,
-		})
-
-		suite.Require().NoError(err)
-		suite.Require().NotNil(res)
+		req := &types.AddressBytesToStringRequest{AddressBytes: address.Bytes()}
+		testdata.DeterministicIterations(suite.ctx, suite.Require(), req, suite.queryClient.AddressBytesToString, 0, true)
+	})
 
-		suite.Require().Equal(res.AddressBytes, prevRes)
-	}
+	req := &types.AddressBytesToStringRequest{AddressBytes: addr.Bytes()}
+	testdata.DeterministicIterations(suite.ctx, suite.Require(), req, suite.queryClient.AddressBytesToString, 0, false)
 }
 
 func (suite *DeterministicTestSuite) TestGRPCQueryAddressStringToBytes() {
 	rapid.Check(suite.T(), func(t *rapid.T) {
 		address := testdata.AddressGenerator(t).Draw(t, "address-string")
-		suite.runStringToAddressBytesIterations(address.String(), address.Bytes())
+
+		req := &types.AddressStringToBytesRequest{AddressString: address.String()}
+		testdata.DeterministicIterations(suite.ctx, suite.Require(), req, suite.queryClient.AddressStringToBytes, 0, true)
 	})
 
-	suite.runStringToAddressBytesIterations(addr.String(), addr.Bytes())
+	req := &types.AddressStringToBytesRequest{AddressString: addr.String()}
+	testdata.DeterministicIterations(suite.ctx, suite.Require(), req, suite.queryClient.AddressStringToBytes, 0, false)
```

Comment:
x/auth's `AddressStringToBytes` query uses 0 gas. it reads from the keeper's bech32Prefix field directly.

Are we fine to still okay to keep this as `module_query_safe`?

Reply:
Looks like you made an issue for this @amaurym? Thanks for that 🙏

Reply:
Oh yeah, forgot to link it here: https://github.com/cosmos/cosmos-sdk/issues/13625